src/main/java/com/configcat/FormattableLogMessage.java

```diff
@@ -0,0 +1,25 @@
+package com.configcat;
+
+class FormattableLogMessage {
+
+    private String cachedMessage;
+    private final String message;
+    private final Object[] args;
+
+    FormattableLogMessage(String message, Object... args) {
+        this.message = message;
+        this.args = args;
+    }
+
+    public String formatLogMessage(){
```

Comment:
nit: As this method is not called from outside, protected accessibility would be a better choice.

```suggestion
    protected String formatLogMessage(){
```